In [38]:
from pathlib import Path

import pandas as pd

In [39]:
# Define the location of the raw data folder
RAW_DATA_DIR = Path.cwd().parent / "data" / "raw"

# Load the datasets
neutral_df = pd.read_csv(RAW_DATA_DIR / "Neutral.csv", encoding="latin1")

depression_df = pd.read_csv(RAW_DATA_DIR / "Depression.csv", encoding="latin1")

suicidal_df = pd.read_csv(
    RAW_DATA_DIR / "Suicadal_tendencies_data.csv", encoding="latin1"
)

In [40]:
# Explore dataset 1 to see compatibility before merging
neutral_df.head()

,Label,TEXT
0,0.0,TIL the movie Starship Troopers was never adap...
1,0.0,What do you call a fat baby?
2,0.0,Two morons are sitting on a fence. The big one...
3,0.0,I covered all my weapons in glue.
4,0.0,Joke I made up: Caveman and a bear walk into a...


In [41]:
# Explore dataset 2 to see compatibility before merging
depression_df.head()

,Label,TEXT
0,1,Fell in love with a girl to the point of exhau...
1,1,I was once a kind romantic who strove to help ...
2,1,! I wake up every day thinking why am I living...
3,1,I have problems with my girlfriend. I cannot b...
4,1,"My whole life is one continuous hell, in which..."


In [42]:
# Explore dataset 3 to see compatibility before merging
suicidal_df.head()

,Label,TEXT
0,2,I don't know what to do with my life
1,2,I think it's tonight.
2,2,Day 1: Realizations
3,2,I'm writing a story were somebody is talked ou...
4,2,Idk. Help. I guess.


In [43]:
# Validate data, check the dimensions
print("Neutral:", neutral_df.shape)
print("Depression:", depression_df.shape)
print("Suicidal:", suicidal_df.shape)

Neutral: (6074, 2)
Depression: (4577, 2)
Suicidal: (7066, 2)


In [44]:
# Check the column names
print(neutral_df.columns)
print(depression_df.columns)
print(suicidal_df.columns)

Index(['Label', 'TEXT'], dtype='str')
Index(['Label', 'TEXT'], dtype='str')
Index(['Label', 'TEXT'], dtype='str')


### Observations

All three datasets contained the expected Label and TEXT columns after cleaning.

The Depression dataset originally contained an extra column named `Unnamed: 2`. Inspection showed the column contained only a single non-null value and was unrelated to the analysis, so it was removed before merging the datasets.

In [45]:
# Check for missing values
print("Neutral")
print(neutral_df.isnull().sum())

print("\nDepression")
print(depression_df.isnull().sum())

print("\nSuicidal")
print(suicidal_df.isnull().sum())

Neutral
Label    4181
TEXT        9
dtype: int64

Depression
Label    0
TEXT     4
dtype: int64

Suicidal
Label    0
TEXT     5
dtype: int64


In [46]:
# Fill missing labels in the Neutral dataset with 0
neutral_df["Label"] = neutral_df["Label"].fillna(0)

In [47]:
neutral_df.isnull().sum()
neutral_df["Label"].value_counts()

Label
0.0    6074
Name: count, dtype: int64

In [ ]:
# Confirmed 9 missing text records are still there.
neutral_df.isnull().sum()

Label    0
TEXT     9
dtype: int64

### Observations

Missing values were found only in the TEXT column.

Rows containing missing text were removed because machine learning models cannot learn from records that do not contain textual information.

No missing values were present in the Label column after validation.

In [49]:
# Merge the datasets
mental_health_df = pd.concat(
    [neutral_df, depression_df, suicidal_df], ignore_index=True
)

In [50]:
# Verify merge
print(mental_health_df.shape)

(17717, 2)


### Observations

After validation and cleaning, the three datasets were merged into a single DataFrame.

The merged dataset contains examples from each mental health category while preserving the original class labels.

In [51]:
# Check missing values in merged dataset
mental_health_df.isnull().sum()

Label     0
TEXT     18
dtype: int64

In [52]:
# Remove rows with missing text
mental_health_df = mental_health_df.dropna(subset=["TEXT"])

In [53]:
# Check missing values
mental_health_df.isnull().sum()

Label    0
TEXT     0
dtype: int64

In [54]:
# Check class distribution
mental_health_df["Label"].value_counts()

Label
2.0    7061
0.0    6065
1.0    4573
Name: count, dtype: int64

### Observations

The merged dataset contains three classes:

- Neutral: 6,065 records
- Depression: 4,573 records
- Suicidal Tendencies: 7,061 records

The classes are reasonably balanced, although the Depression class contains fewer observations than the other two categories. This level of imbalance is relatively small and should still allow supervised classification models to learn meaningful patterns.

### Data Quality Summary

The dataset required minimal cleaning prior to analysis.

Data quality issues included:

- One unnecessary column in the Depression dataset.
- Missing text values in each dataset.
- Character encoding requiring the use of `encoding="latin1"` when importing the CSV files.

After these issues were addressed, the merged dataset was suitable for text preprocessing and machine learning.

In [55]:
# Save final dataset
mental_health_df.to_csv("../data/processed/mental_health_text_dataset.csv", index=False)

In [ ]:
# Check for corrupted characters
mental_health_df[mental_health_df["TEXT"].str.contains("Ã", na=False)]

,Label,TEXT
3811,0.0,what�s hot and new?
3841,0.0,sad day!!!!! Love you Ma� (8 years ago!)
3990,0.0,28� in VanCity? Holy shizz. I will be hom...
4311,0.0,"download iEmoticons from the AppStore. 0,79ï..."
4522,0.0,"Coool. - Weekend wasn�t that great, eeeh...."
4571,0.0,">>i don�t live in UK, so I can�t sa..."
4697,0.0,I painted my nails metallic blue but I�m ...
4840,0.0,?� nh?n ???c bonus r?i. Ch?c ch? ?? bï¿...
4859,0.0,Looking for a new SMS sending solution? http:...
4860,0.0,Eat Sleep Draw!: Part of my Neuroanatomy take-...
